worker/uniter/modes.go

```diff
@@ -374,8 +418,8 @@ func modeContext(name string, err *error) func() {
 
 // continueAfter is commonly used at the end of a Mode func to execute the
 // operation returned by creator and return ModeContinue (or any error).
-func continueAfter(u *Uniter, creator creator) (Mode, error) {
-	if err := u.runOperation(creator); err != nil {
+func continueAfter(u *Uniter, doing string, creator creator) (Mode, error) {
```

Comment:
I am bothered by this -- if we're performing some operation, isn't it up to the operation to know what we're doing?